drivers/hyperv/hyperv.go

```diff
@@ -186,17 +190,30 @@ func (d *Driver) PreCreateCheck() error {
 		return err
 	}
 
-	// Downloading boot2docker to cache should be done here to make sure
+	// Downloading boot2docker/windows-server to cache should be done here to make sure
 	// that a download failure will not leave a machine half created.
 	b2dutils := mcnutils.NewB2dUtils(d.StorePath)
-	err = b2dutils.UpdateISOCache(d.Boot2DockerURL)
+
+	if mcnutils.ConfigGuestOSUtil.GetGuestOS() != "windows" {
+		err = b2dutils.UpdateISOCache(d.Boot2DockerURL)
+	} else {
+		err = b2dutils.UpdateVHDCache(d.WindowsVHDUrl)
+	}
+
 	return err
 }
 
 func (d *Driver) Create() error {
 	b2dutils := mcnutils.NewB2dUtils(d.StorePath)
-	if err := b2dutils.CopyIsoToMachineDir(d.Boot2DockerURL, d.MachineName); err != nil {
-		return err
+
+	if mcnutils.ConfigGuestOSUtil.GetGuestOS() == "windows" {
+		if err := b2dutils.CopyWindowsIsoToMachineDir(d.WindowsVHDUrl, d.MachineName); err != nil {
```

Comment:
nit: now that we switched to vhds perhaps we should replace ISO with VHD

Reply:
This has been resolved!